cpp/src/io/text/multibyte_split.cu

```diff
@@ -383,17 +387,31 @@ std::unique_ptr<cudf::column> multibyte_split(cudf::io::text::data_chunk_source
     stream,
     streams);
 
+  // String offsets point to the first character of a field
+  // This finds the first field whose first character starts inside or after the byte range
   auto relevant_offsets_begin = thrust::lower_bound(rmm::exec_policy(stream),
                                                     string_offsets.begin(),
                                                     string_offsets.end() - 1,
                                                     byte_range.offset());
 
-  auto relevant_offsets_end = thrust::upper_bound(rmm::exec_policy(stream),
+  // This finds the first field beginning after the byte range.
+  // We shift it by 1 to also copy this last offset
+  auto relevant_offsets_end = thrust::lower_bound(rmm::exec_policy(stream),
                                                   string_offsets.begin(),
                                                   string_offsets.end() - 1,
                                                   byte_range.offset() + byte_range.size()) +
                               1;
```

Comment:
This is very unfortunate formatting. Perhaps change it to `1 + thrust::lower_bound...`?